### ISSUE TITLE
Support learning time series data in variant length

### ISSUE THREAD

**Issue by tomfisher** [rtavenar/tslearn#16]
Hi,

One of important features that make time series learning different from generic data is its invariant length, which may also make tslearn unique. 

It would be really appreciated that there is a specific paragraph elaborating which length-variant algorithms can be used in tslearn. For shapelets example, the related demo is about how to deal with equal length data although the algorithm itself supports TS in variant length.

Thanks

**Comment by rtavenar** [comment]
Hi,

It's a good idea to have such a page (listing algorithms working on variable-length time series) in the doc.
Concerning the Shapelet module, it is based on keras, and I don't know how to deal with TS of different length (except for recurrent units) in keras. If anyone can help on this, feel free!

**Comment by rtavenar** [comment]
Concerning the doc, I added a page that is supposed to list all methods that can operate on datasets composed of time series of diverse lengths (http://tslearn.readthedocs.io/en/latest/variablelength.html). 

For the second point (shapelet module), it is likely to take some time before a fix is found (except if a keras master takes a look at it :)

Best,
Romain

**Comment by tomfisher** [comment]
I saw the commit about this. It could be really useful if an example is given there :)

**Comment by njumarlon** [comment]
Yes. Could you give an example that illustrates how to perforn the methods on variable-length time series? I tried to find it on the page about variable-length time series but couldn't figure out how to do it.
